### PR TITLE
Add umsg ioctl test cases. 

### PR DESCRIPTION
**Pull request by jelonanX**

### CODE REVIEW

File: testing/xfpga/test_umsg_c.cpp [hunk 1]
```diff
@@ -194,6 +226,47 @@ TEST_P(umsg_c_p, set_umsg_attr_ioctl_err) {
...
 }
+
+
+TEST_P(umsg_c_p, get_umsg_ptr_ioctl_err) {
+  uint64_t *value = 0;
+
+  // register an ioctl handler that will return -1 and set errno to EINVAL
+  system_->register_ioctl_handler(FPGA_PORT_UMSG_ENABLE, dummy_ioctl<-1,EINVAL>);
+  system_->register_ioctl_handler(FPGA_PORT_DMA_UNMAP, dummy_ioctl<-1,EINVAL>);
+  EXPECT_EQ(FPGA_INVALID_PARAM, xfpga_fpgaGetUmsgPtr(handle_, &value));
```
Comment (r-rojo) [inline testing/xfpga/test_umsg_c.cpp]: Are two ioctl calls done inside this function?

Comment (jelonanX) [reply testing/xfpga/test_umsg_c.cpp]: From what I can tell through gdb, two+ ioctl calls can happen through calling fpgaGetUmsgPtr

File: testing/xfpga/test_umsg_c.cpp [hunk 2]
```diff
@@ -348,41 +421,36 @@ TEST_P(umsg_c_p, test_umsg_drv_06) {
...
 
-	////////////////////////////////////////
-	// Disable the following lines because it 
-	// modifies handle to gain coverage.
-	////////////////////////////////////////
-
-  //// Invalid Magic Number
-  //ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &h, 0));
+  // Invalid Magic Number
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &handle_, 0));
```
Comment (tswhison) [inline testing/xfpga/test_umsg_c.cpp]: handle_ is already opened from SetUp.  Previous handle_ value will be leaked.

File: testing/xfpga/test_umsg_c.cpp [hunk 3]
```diff
@@ -132,6 +164,14 @@ class umsg_c_p
 
   virtual void TearDown() override {
     EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+
```
Comment (tswhison) [inline testing/xfpga/test_umsg_c.cpp]: handle_ is being opened in the SetUp, but isn't being closed here in TearDown.  Additionally, in some tests (test_umsg_drv_07), handle_ is open'ed again.  I think we have some resource leaks and potential memory errors that need to be addressed.

File: testing/xfpga/test_umsg_c.cpp [hunk 4]
```diff
@@ -283,39 +359,34 @@ TEST_P(umsg_c_p, test_umsg_drv_04) {
...
 
-  // _handle->fddev = fddev;
-  // EXPECT_EQ(FPGA_OK, xfpga_fpgaClose(handle_));
+  _handle->fddev = fddev;
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaClose(handle_));
 
-  // // Invlaid Input Paramter
-  // ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tok, &h, 0));
+  // Invlaid Input Paramter
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &handle_, 0));
```
Comment (tswhison) [inline testing/xfpga/test_umsg_c.cpp]: If this call is successful, then the previous handle_ value is leaked.

File: testing/xfpga/test_umsg_c.cpp [hunk 5]
```diff
@@ -348,41 +421,33 @@ TEST_P(umsg_c_p, test_umsg_drv_06) {
...
 
-  //_handle->fddev = fddev;
-  //EXPECT_EQ(FPGA_OK, xfpga_fpgaClose(h));
+  _handle->fddev = fddev;
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaClose(handle_));
 
-  //// Invalid Input Parameter
-  //ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokend_[0], &handle_, 0));
+  // Invalid Input Parameter
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &handle_, 0));
```
Comment (tswhison) [inline testing/xfpga/test_umsg_c.cpp]: If this call is successful, then the previous handle_ value is leaked.

**Pull request merged**